workflows/mark03.smk

```diff
@@ -452,3 +465,240 @@ rule combined_logProb_iteration_plot:
+            input.append(
+                "{DATADIR}/mark03/analysis/MCMC_"
+                + str(mcmc_seed)
+                + "-{mutation_data_id}-"
+                + "iT_m_{n_nodes}_"
+                + str(n_mcmc_tree_moves)
+                + "_"
+                + str(mcmc_move_seed)
+                + "_oT_{base_tree_type}_{n_nodes,\d+}_{base_tree_seed,\d+}"
+                + "-{mcmc_config_id}"
+                + "/T_{base_tree_type}_{n_nodes,\d+}_{base_tree_seed,\d+}/log_prob.json"
+            )
+
+        # all other cases
+        else:
+            input.append(
+                "{DATADIR}/mark03/analysis/MCMC_"
+                + str(mcmc_seed)
+                + "-{mutation_data_id}-iT_"
+                + str(init_tree_type)
+                + "_{n_nodes,\d+}_"
+                + str(init_tree_seed)
+                + "-{mcmc_config_id}/T_{base_tree_type}_{n_nodes,\d+}_{base_tree_seed,\d+}/log_prob.json"
+            )
+    return input
+
+
+rule combined_logProb_iteration_plot_noHuntress:
+    """Make combined logProb iteration plot. - excludes huntress"""
+    input:
+        # calls analyze_metric rule
+        all_chain_logProb=make_combined_log_prob_iteration_in_noHuntress(),
+    output:
+        combined_logP_iter="{DATADIR}/{experiment}/plots/{mcmc_config_id}/{mutation_data_id}/T_{base_tree_type}_{n_nodes,\d+}_{base_tree_seed,\d+}/log_prob_iter_noHuntress.svg",
+    run:
+        # load the data
+        logP_chains = []
+        # get the initial tree type, same order as the input
+        initial_tree_type = make_combined_metric_iteration_in()[1]
+        # for each chain
+        for each_chain_metric in input.all_chain_logProb:
+            # load the distances
+            _, logP = yg.serialize.read_metric_result(Path(each_chain_metric))
+            # append to the list
+            logP_chains.append(logP)
+
+            # Create a figure and axis
+        fig, ax = plt.subplots()
+
+        # Define the list of colors to repeat
+        colors = {
+                "s": "green",
+                "d": "blue",
+                "r": "orange",
+                "mcmc": "purple",
+            }
+
+        labels = {
+            "s": "Star",
+            "d": "Deep",
+            "r": "Random",
+            "mcmc": "MCMC5",
+        }
+
+        # Define opacity and line style
+        alpha = 0.6
+        line_style = "solid"
+
+        # Plot each entry of distance chain as a line with a color unique to the
+        # initial tree type onto one axis
+        for i, logP in enumerate(logP_chains):
+            color = colors[initial_tree_type[i]]
+            ax.plot(
+                logP,
+                color=color,
+                label=f"{labels[initial_tree_type[i]]}",
+                alpha=alpha,
+                linestyle=line_style,
+            )
+
+            # Set labels and title
+        ax.set_ylabel(f"Log Probability:" + r"$\log(P(D|T,\theta))$")
+        ax.set_xlabel("Iteration")
```

Comment:
Repeated code block - factor this out into a function. Open an issue for this and do later.

Reply:
See issue
- #121